models/plays_by_pageview.sql

```diff
@@ -0,0 +1,140 @@
+{{ 
+  config(
+    materialized='table',
+    unique_key = 'play_id',
+    sort = 'start_tstamp',
+    dist = 'play_id'
+  )
+}}
+
+with prep as (
+
+  select
+    {{ dbt_utils.hash('i.page_view_id||i.media_id') }} play_id,
+    i.page_view_id,
+    i.media_id,
+    max(i.domain_sessionid) as domain_sessionid,
+    i.domain_userid,
+    i.duration,
+    i.title,
+    i.media_type,
+    i.media_player_type,
+    i.page_referrer,
+    i.content_url,
+    i.geo_region_name,
+    i.br_name,
+    i.dvce_type,
+    i.os_name,
+    i.os_timezone,
+    coalesce(sum(p.weight_rate * i.duration), 0)::int as play_time_sec_estimated,
+    min(start_tstamp) start_tstamp,
+    sum(case when i.event_type in ('seek', 'seeked') then 1 else 0 end) as seeks,
+    max(case 
+          when i.event_type = 'ended' then 100 
+          when i.percent_progress = 0 then NULL 
+          else i.percent_progress 
+        end) as last_percent_progress,
+    max(i.end_tstamp) end_tstamp,
+    sum(i.play_time_sec_amended) as play_time_sec,
+    sum(case when i.muted then i.play_time_sec_amended else 0 end) as play_time_sec_muted,
+    coalesce(sum(i.playback_rate * i.play_time_sec) / nullif(sum(i.play_time_sec), 0), 0) as avg_playback_rate,
+    coalesce(sum(i.play_time_sec_amended)/ nullif(max(i.duration), 0), 0) as retention_rate
+  
+  from {{ ref("interactions") }} i
+
+  left join {{ ref("pivot_base") }} p
+  on i.percent_progress = p.percent_progress
+
+  group by
+    {{ dbt_utils.hash('page_view_id||media_id') }},
+    page_view_id,
+    media_id,
+    domain_userid,
+    duration,
+    title,
+    media_type,
+    media_player_type,
+    page_referrer,
+    content_url,
+    geo_region_name,
+    br_name,
+    dvce_type,
+    os_name,
+    os_timezone
+
+  having sum(case when i.event_type in ('play', 'playing') then 1 else 0 end) > 0
```

Comment:
It should have probably been called `plays` instead of `views` as I only meant to count the number of times the users actually played a video when visiting the page instead of the number of page views with a video embedded.